csharp/src/Apache.Arrow/Ipc/ArrowStreamReaderImplementation.cs

```diff
@@ -78,23 +79,24 @@ protected async ValueTask<RecordBatch> ReadRecordBatchAsync(CancellationToken ca
             }
 
             RecordBatch result = null;
-            await Buffers.RentReturnAsync(messageLength, async (messageBuff) =>
+            await ArrayPool<byte>.Shared.RentReturnAsync(messageLength, async (messageBuff) =>
             {
                 int bytesRead = await BaseStream.ReadFullBufferAsync(messageBuff, cancellationToken)
                     .ConfigureAwait(false);
                 EnsureFullRead(messageBuff, bytesRead);
 
                 Flatbuf.Message message = Flatbuf.Message.GetRootAsMessage(CreateByteBuffer(messageBuff));
 
-                await Buffers.RentReturnAsync((int)message.BodyLength, async (bodyBuff) =>
-                {
-                    int bodyBytesRead = await BaseStream.ReadFullBufferAsync(bodyBuff, cancellationToken)
-                        .ConfigureAwait(false);
-                    EnsureFullRead(bodyBuff, bodyBytesRead);
+                int bodyLength = checked((int)message.BodyLength);
+
+                IMemoryOwner<byte> bodyBuffOwner = _allocator.Allocate(bodyLength);
```

Comment:
again: is it acceptable to have `bodyBuffOwner` be a null?

what is the outcome of reading into an empty memory?

Reply:
What else would you expect to be returned when `0` gets passed into `Allocate`? Is allocating an object with an empty Memory worth it?

Reply:
I asked this question because I did not understand why it's returning a null, not because I thought it's wrong to do so.